### PR TITLE
fixed error where footprint fails to generate on some map projected images. fixes #4390

### LINKED ISSUE #4390: Unable to generate footprint with unknown GEOS exception
**ISIS version(s) affected**: 4.3.0

**Description**  
When attempting to generate a footprint on a L2 product using `footprintinit from=/scratch/jlaura/europa_mosaics/GLL_Obs/equi/33ESGLOCOL01_GalileoSSI_Equi.cub` I am seeing the following error:

```
**USER ERROR** Cannot generate polygon for [33ESGLOCOL01_GalileoSSI_Equi.cub].
**ERROR** Unable to create image footprint (Fix360Poly) due to unknown exception.
```
It does not seem to matter what linc/sinc I select. Due to the geometry of many of the other images (and the small size), I have been using `sinc=1` and `linc=1`.

**How to reproduce**  
Data are internal at this point and are staged here:
`footprintinit from=/scratch/jlaura/europa_mosaics/GLL_Obs/equi/33ESGLOCOL01_GalileoSSI_Equi.cub`

Here are some other images with identical behavior:
- C9ESGLOBAL01_GalileoSSI_Equi.cub
- 33ESGLOCOL01_GalileoSSI_Equi.cub  
- G2ESPHOTOM01_GalileoSSI_Equi.cub
- ../npola/10ESGLOBAL01_GalileoSSI_NPola.cub
- ../spola/25ESGLOBAL01_GalileoSSI_SPola.cub

**Possible Solution**  
???

**Additional context**  
I can see that the catch block that is throwing starts [here on line 1333](https://github.com/USGS-Astrogeology/ISIS3/blob/aab0e63088046690e6c031881825596c1c2cc380/isis/src/base/objs/ImagePolygon/ImagePolygon.cpp#L1333), but the fact that this is an unknown exception really is not helpful.

### PR DESCRIPTION
**Pull request by Kelvinrr**
## Description

Fixes bug where footprints fail to generate when these specific conditions are met: 

1. Input image is map projected 
2. target crosses the 0/360 boundary
3. some line/sample in the image projects to the polar lat lons: (90,0) and/or (-90/0) 
4. that line sample that projects to (90,0)/(-90, 0) does not project onto the body/is a null pixel. 

Which results in the image footprint becoming mangled. 

## Related Issue

https://github.com/USGS-Astrogeology/ISIS3/issues/4390

## Motivation and Context

Some map projected images were not generating valid footprints. 

## How Has This Been Tested?

Original image in the issue successfully generated it's footprint when it previous excepted with incomplete linear ring coordinates. 

## Screenshots (if appropriate):

using image: `/archive/projects/europa/GLL_FinProducts/*/33ESGLOCOL01_GalileoSSI_Equi.cub`

Footprint trying to cross the pole, had to use a web geojson app since I had incomplete coords I was just printing out: 

Screen Shot 2022-04-11 at 2 48 44 PM

After fix, viewed in qmos:

Screen Shot 2022-04-14 at 10 19 35 PM

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/ImagePolygon/ImagePolygon.cpp [hunk 1]
```diff
@@ -1329,6 +1330,11 @@ namespace Isis {
       msg += "isis operation exception [" + IString(e.what()) + "]";
       throw IException(e, IException::Unknown, msg, _FILEINFO_);
     }
+    catch(std::exception &e) {
+      std::string msg = "Caught std::exception: ";
+      msg += e.what();
+      throw IException(IException::Unknown, msg, _FILEINFO_); 
+    }
```
Comment (Kelvinrr) [inline isis/src/base/objs/ImagePolygon/ImagePolygon.cpp:1337]: std::exception wasn't being caught causing the real error of an incomplete linestring to be swllowed by `catch(...)`

File: isis/src/base/objs/ImagePolygon/ImagePolygon.cpp [hunk 2]
```diff
@@ -852,7 +850,8 @@ namespace Isis {
 
       if (nPoleSample >= 0.5 && nPoleLine >= 0.5 &&
          nPoleSample <= p_cube->sampleCount() + 0.5 &&
-         nPoleLine <= p_cube->lineCount() + 0.5) {
+         nPoleLine <= p_cube->lineCount() + 0.5 &&
+         SetImage(nPoleSample, nPoleLine)) {
```
Comment (Kelvinrr) [inline isis/src/base/objs/ImagePolygon/ImagePolygon.cpp:854]: SetImage checks to see if the pixel can get a valid projection when going image->ground. 

idk why ground->image successfully returns pixel coords but image->ground acknowledges that there is no intersection?

**Review by jlaura — APPROVED** [review]
I'm happy to build a dev version and test on the whole archive as well once this is merged.

**Comment by Kelvinrr** [comment]
gonna wait until test finishes

**Comment by acpaquette** [comment]
This should be good, only tests failing are the ones that have been fixed

**Pull request merged**